gst/nnstreamer/elements/gsttensor_demux.c

```diff
@@ -511,15 +513,20 @@ gst_tensor_demux_chain (GstPad * pad, GstObject * parent, GstBuffer * buf)
 
       num = g_strv_length (strv);
       for (j = 0; j < num; j++) {
-        gint64 idx = g_ascii_strtoll (strv[j], NULL, 10);
-        mem = gst_buffer_get_memory (buf, idx);
-        gst_buffer_append_memory (outbuf, mem);
+        idx = (guint) g_ascii_strtoll (strv[j], NULL, 10);
+        _info =
+            gst_tensors_info_get_nth_info (&tensor_demux->tensors_config.info,
+            idx);
+        mem = gst_tensor_buffer_get_nth_memory (buf, idx);
+        gst_tensor_buffer_append_memory (outbuf, mem, _info);
       }
       g_strfreev (strv);
       list = list->next;
     } else {
-      mem = gst_buffer_get_memory (buf, i);
-      gst_buffer_append_memory (outbuf, mem);
+      _info =
+          gst_tensors_info_get_nth_info (&tensor_demux->tensors_config.info, i);
+      mem = gst_tensor_buffer_get_nth_memory (buf, i);
+      gst_tensor_buffer_append_memory (outbuf, mem, _info);
```

Comment:
The input argument of each function is checked for null inside the function. 
But what about checking the return value of "gst_tensor_buffer_append_memory" ?
There seem to be many cases where the function returns false.

Reply:
If the value inside info entered in this function is always vaild, it's okay.

Reply:
Updated, plz see the changes again.